pages/404.js

```diff
@@ -4,10 +4,18 @@ import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 import Link from "next/link";
 import { ReportAProblem } from "../components/organisms/ReportAProblem";
 import { ActionButton } from "../components/atoms/ActionButton";
+import { useEffect } from "react";
 
 export default function error404(props) {
   const { t } = useTranslation("common");
 
+  process.env.NEXT_PUBLIC_ADOBE_ANALYTICS_URL
+    ? useEffect(() => {
+        window.adobeDataLayer = window.adobeDataLayer || [];
```

Comment:
So I was curious about this and decided to do some reading and I think we should avoid useEffects conditionally but rather put the condition _inside_ the useEffect. Some details here:

https://reactjs.org/docs/hooks-rules.html

Reply:
For all I know, it could be the reason that it wasn't working in the top-level app.js